pwa-shell/src/ui/content-detail/ContentDetail.js

```diff
@@ -4,20 +4,51 @@ import PropTypes from 'prop-types';
 import { Container } from 'reactstrap';
 import ProtectedContentLoginContainer from 'ui/login/ProtectedContentLoginContainer';
 import ContentDetailTopBarContainer from 'ui/content-detail/ContentDetailTopBarContainer';
+import SwipeContentNavigator from 'ui/common/SwipeContentNavigator';
 import PageContainer from 'ui/common/PageContainer';
 import ItemCategoryListContainer from 'ui/common/ItemCategoryListContainer';
 
 class ContentDetail extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      nextURL: '',
+      previousURL: '',
+    };
+  }
   componentDidMount() {
-    this.props.fetchContentDetail();
+    const { location, match } = this.props;
+    this.props.fetchContentDetail(location, match.params);
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
`componentWillReceiveProps` is deprecated. Can you make it work using `componentDidUpdate`?